Detectors/FIT/FT0/simulation/src/Digits2Raw.cxx

```diff
@@ -99,36 +111,11 @@ void Digits2Raw::readDigits(const std::string fileDigitsName)
     for (int ibc = 0; ibc < nbc; ibc++) {
       auto& bcd = digitsBC[ibc];
       intRecord = bcd.getIntRecord();
-      int nHBF = mSampler.fillHBIRvector(HBIRVec, lastIR, intRecord);
-      lastIR = intRecord + 1;
-      if (nHBF) {
-        for (int j = 0; j < nHBF - 1; j++) {
-          o2::InteractionRecord rdhIR = HBIRVec[j];
-          for (int link = 0; link < (int)mPages.size(); ++link) {
-            setRDH(mPages[link].mRDH, link, rdhIR);
-            mPages[link].flush(mFileDest);
-          }
-        }
-
-        uint32_t current_orbit = intRecord.orbit;
-        if (old_orbit != current_orbit) {
-          for (DataPageWriter& writer : mPages)
-            writer.flush(mFileDest);
-          for (int nlink = 0; nlink < NPMs; ++nlink)
-            setRDH(mPages[nlink].mRDH, nlink, intRecord);
-          old_orbit = current_orbit;
-        }
-        auto channels = bcd.getBunchChannelData(digitsCh);
-        int nch = channels.size();
-        if (nch) {
-          convertDigits(bcd, channels, lut, intRecord);
-        }
-      }
+      auto channels = bcd.getBunchChannelData(digitsCh);
+      if (!channels.empty())
```

Comment:
This line does nothing, is it supposed to prevent calling ``convertDigits``?

Reply:
yes, if no one channel fired

Reply:
but this is not what it does: to not call ``convertDigits for empty channels you should do
````
if (!channels.empty()) {
  convertDigits(bcd, channels, lut, intRecord);
}
````

Reply:
And this one?